src/dg/dg.cpp

```diff
@@ -2060,14 +2464,20 @@ void DGBase<dim,real,MeshType>::allocate_system ()
     right_hand_side.add(1.0); // Avoid 0 initial residual for output and logarithmic visualization.
     dual.reinit(locally_owned_dofs, ghost_dofs, mpi_communicator);
 
+    // Allocate for auxiliary equation only.
+    allocate_auxiliary_equation ();
```

Comment:
should this only be done when `use_auxiliary_eq == true` ?

Reply:
Yes, but I have it doing that within DGStrong. So for DG weak it does nothing but in DG strong it then gets that "use_auxiliary_eq" flag and allocates.

Reply:
My apologies, my answer above was wrong. For DG strong it always has to allocate the auxiliary solution, whether or not there is dissipation because the residual functions fetch the auxiliary modal coefficients. It does not necessarily have to allocate the RHS vector when there isn't dissipation, but it does need to allocate the auxiliary solution.